portfolio/src/main/java/com/google/sps/data/VideoAnalysis.java

```diff
@@ -0,0 +1,52 @@
+package com.google.sps.data;
+import com.google.sps.data.Query;
+import com.google.sps.data.SentimentTools;
+import com.google.sps.data.Video;
+
+// Represents a video analysis, holds the results from an analysis and can also analyze a video that is passed in.
+public class VideoAnalysis {
+    private float sentimentScore;
+    private float sentimentMagnitude;
+    private String query;
+    private static int QUERY_SIZE = 7;
+    
+    //If given a video we will run analysis on it !!THIS IS SLOWER THAN JUST PASSING IN THE VALUES IF YOU HAVE THEM!!
+    public VideoAnalysis(Video video) {
+        SentimentTools sentimentAnalysis = new SentimentTools(video);
+        Query queryAnalysis = new Query(video, QUERY_SIZE);
+        this.sentimentScore = sentimentAnalysis.getScore();
+        this.sentimentMagnitude = sentimentAnalysis.getMagnitude();
+        this.query = queryAnalysis.toString();
+    }
+
+    //Pass in the results from a Video analysis to create this object
+    public VideoAnalysis(float _sentimentScore, float _sentimentMagnitude, String _query) {
+        this.sentimentScore = _sentimentScore;
+        this.sentimentMagnitude = _sentimentMagnitude;
+        this.query = _query;
+    }
+
```

Comment:
Why do we need setters?

Reply:
Agree, they might not be necessary here.

Reply:
figured they would be useful for when we wanted to refresh the analysis

Reply:
True, they might be. That should be included in that PR, though. In general, we want to add as little code that gets the job done.